pkg/envoy/admin/tls/pki.go

```diff
@@ -73,10 +73,14 @@ func GenerateClientCert(ca tls.Certificate) (util_tls.KeyPair, error) {
 	return util_tls.NewCert(*rootCert, ca.PrivateKey.(*rsa.PrivateKey), ClientCertSAN, util_tls.ClientCertType, util_tls.DefaultKeyType, ClientCertSAN)
 }
 
-func GenerateServerCert(ca tls.Certificate, ip string) (util_tls.KeyPair, error) {
+func GenerateServerCert(ca tls.Certificate, address string, advertisedAddress string) (util_tls.KeyPair, error) {
```

Comment:
Can we change the signature to be:
```go
func GenerateServerCert(ca tls.Certificate, ips ...string) (util_tls.KeyPair, error)
```
I think it's better if the function won't be aware of the "advertised" concept.

Reply:
Ok, but how do you define the commonName required by `util_tls.NewCert()` ?

Reply:
Actually, it seems like we don't have to use CommonName. I checked, starting Go 1.15 `http.Client` is going to fail the request if there is only CommonName. When there are both SAN and CN only the SANs are being used for validation.

Reply:
What do you expect me to do here ? Remove commonName from `util_tls.NewCert` function ?

Reply:
Good point, modifying `util_tls.NewCert` is a bigger change and should be done in a separate PR. Could you please pass `hosts[0]` or `""` as a `commonName` for NewCert?

Reply:
Ok ! It’s done !